src/data_gradients/feature_extractors/object_detection/similarity.py

```diff
@@ -0,0 +1,225 @@
+from typing import List, Optional
+
+import numpy as np
+from torch.nn import Identity
+from torchvision.models import swin_v2_b, Swin_V2_B_Weights
```

Comment:
I think we should add `torchvision` to requirements.txt with version specifier `>=` corresponding to a version where Swin was introduced.

Reply:
100 percent, totally forgot.
Thanks!

Reply:
b64828c3ce5e111def6aa32e59a76024cd3a6d62